### ISSUE TITLE
Undefined behavior while decoding characters not in the base62 alphabet

### ISSUE THREAD

**Issue by jchampemont** [seruco/base62#2]
When trying to decode something that contains invalid base62 characters, the behavior is undefined and not always the same:

- with ascii characters, decoding works but the result is total garbage
- with more complex utf-8 characters, an exception is thrown:
```
java.lang.ArrayIndexOutOfBoundsException: -16

	at io.seruco.encoding.base62.Base62.translate(Base62.java:84)
```
Code that triggered this case:
```
Base62.createInstance().decode("😱".getBytes());
```

That might be critical when the encoded string is user/external input.

I see plenty of possible solutions here, and not sure which one you prefer. If you pick one I am willing to make a PR.

- Do as JDK Base64: throw IllegalArgumentException if the string is not valid according to the chosen alphabet
- Implement a method to check if a string is valid according to the chosen alphabet
- Publicly expose the alphabets to allow callers to make their own checks.

WDYT?

**Comment by tbenbrahim** [comment]
+1 for throw IllegalArgumentException
you would be writing a method to check if a string valid, so making it public rather than private would help those who prefer to avoid the exception.

**Comment by seruco** [comment]
I agree that throwing an IllegalArgumentException is probably the best choice. A PR is highly appreciated! 😊 Thanks for the feedback!

**Comment by seruco** [comment]
Fixed with version `0.1.3`.
